dataprofiler/data_readers/csv_data.py

```diff
@@ -103,24 +103,152 @@ def _check_and_return_options(self, options):
+                        "type": cell_type,
+                        "size": size_of_cell
+                    })
+                proposed_dataset.append(proposed_row)
+
+            col_types = {}            
+            prior_col_count = None
+            incorrect_delimiter_flag = False
+            cell_type_safe_flag = True
+            cell_end_safe_flag = False
+            cell_start_safe_flag = False
+
+            proposed_delim_type = data_utils.detect_cell_type(proposed_delim)
+        
+            # Reverse to start at bottom
+            # Fewer columns are okay, as long as earlier in file
+            proposed_dataset.reverse()
+            for proposed_row in proposed_dataset:
+
+                # Ensure there's a prior_col_count
+                if not prior_col_count:
+                    prior_col_count = len(proposed_row)
+
+                # Ensure's it's not a description row, then ensures cells consistent
+                if len(proposed_row) != prior_col_count and len(proposed_row) > 1:
+                    incorrect_delimiter_flag = True
+
+
+                # Evalutes each column for potential issues
+                # e.g. '3' being a seperator, while a single row of integers
+                cell_content = []
+                cell_types = []
+                prior_cell_type = None
+                for col_id in range(len(proposed_row)):
+                    proposed_cell = proposed_row[col_id]
+
+                    col_types[col_id] = proposed_cell['type']
+
+                    # Handle if alpha character are seperator                     
+                    # NOTE: delimiter needs two ajoining cells to flag
+                    if proposed_cell['type'] in ['str', 'none'] \
+                       and prior_cell_type in ['str', 'none']:
+                        if proposed_delim.isalpha():
+                            cell_type_safe_flag = False
+                            
+                        # Ensures when spaces occur, there are 3 or more cols
+                        if proposed_delim == ' ' and 2 >= len(proposed_row):
+                            cell_type_safe_flag = False
+
+                    # Handle if integer characters are seperators
+                    # NOTE: delimiter need one adjoining cell to flag
+                    if proposed_delim_type == 'int' and proposed_cell['type']=='int':
+                        cell_type_safe_flag = False
+
+                    # Handle if start or end is always none
+                    if proposed_cell['type']!='none':
+                        if col_id == len(proposed_row)-1:
+                            cell_end_safe_flag = True
+                        if col_id == 0:
+                            cell_start_safe_flag = True
+                    
+                    prior_cell_type = proposed_cell['type']
+
+            edge_safe_flag = False
+            if cell_end_safe_flag or cell_start_safe_flag:
+                edge_safe_flag = True
```

Comment:
This would look cleaner and easier to read if it were just `edge_safe_flag = cell_end_safe_flag or cell_start_safe_flag`